lib/src/entity.dart

```diff
@@ -60,4 +60,23 @@ abstract class Entity extends PositionComponent {
   bool hasBehavior<T extends Behavior>() {
     return findBehavior<T>() != null;
   }
+
+  /// Adds the given [behavior] to this entity only if the entity does not yet
+  /// have the behavior.
+  Future<void> addBehavior<T extends Behavior>(T behavior) async {
+    assert(
+      !hasBehavior<T>(),
+      'The entity already has a behavior of the type $T',
+    );
+    return await super.add(behavior);
+  }
+
+  @override
+  Future<void>? add(Component component) {
```

Comment:
We should add this verification on `addAll` too.

Reply:
It shouldn't be needed, `addAll` delegates to `add` IIRC